solo-chains/runtime/dancelight/src/lib.rs

```diff
@@ -2909,54 +2911,127 @@ impl frame_support::traits::Randomness<Hash, BlockNumber> for BabeCurrentBlockRa
 
 pub struct RemoveParaIdsWithNoCreditsImpl;
 
-impl RemoveParaIdsWithNoCredits for RemoveParaIdsWithNoCreditsImpl {
-    fn remove_para_ids_with_no_credits(
-        para_ids: &mut Vec<ParaId>,
+impl RemoveParaIdsWithNoCreditsImpl {
+    fn charge_para_ids_internal(
```

Comment:
this is becoming hard to read, I would probably split it into different subfunctions that are public in the services payment pallet so that all this code is not in the runtime.

Reply:
also the naming of the function is not ideal, I would call it something like `charge_services_for_para_id`